CommunityToolkit.HighPerformance/Ref{T}.cs

```diff
@@ -2,31 +2,22 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
+#if NET7_0_OR_GREATER
+
 using System.Runtime.CompilerServices;
-using System.Runtime.Versioning;
-#if NETSTANDARD2_1_OR_GREATER
-using System.Runtime.InteropServices;
-#else
-using CommunityToolkit.HighPerformance.Helpers;
-#endif
 
 namespace CommunityToolkit.HighPerformance;
 
 /// <summary>
 /// A <see langword="struct"/> that can store a reference to a value of a specified type.
 /// </summary>
 /// <typeparam name="T">The type of value to reference.</typeparam>
-[RequiresPreviewFeatures(
-    "The Ref<T> type has no compiler support to ensure the lifetime of referenced values is respected, and as such using it incorrectly may lead to GC holes.",
-    Url = "https://github.com/dotnet/runtime/issues/46104")]
 public readonly ref struct Ref<T>
```

Comment:
How about you make this and `ReadOnlyRef<T>` obsolete? I don't think there is a reason for anyone to use it over regular `ref` fields. Especially since you made it exclusive to .NET 7.

Reply:
There's still valid use case scenarios, such as:

- Returning refs to refs. Eg. with this you can have an `out Ref<T>` parameter, which is not otherwise expressible
- I plan on adding more APIs (eg. indexers, etc.) to allow working with GC ref arithmetic easier